### PR TITLE
Prepare for release

### PR DESCRIPTION
**Pull request by omad**
### Reason for this pull request
Grab bag of changes before rolling a new release.

Sorry this has grown so big. I'm not 100% sure everything I want is here, but, it should be merged and reviewed before it gets any more out of hand.

### Proposed changes
- Merge changes from `release-1.5` branch.
- Fixes for lots of warnings in the documentation
- Small fixes to ingestion, include the full path to the configuration file instead of expecting a hard coded version number in the file.
- More RTD documentation on CLI apps
- Include most of the changes in the `whats_new` page
- Only include legacy packages when running legacy python

 - [ ] Closes #xxxx
 - [ ] Tests added / passed
 - [ ] Fully documented, including `docs/about/whats_new.rst` for all changes

### CODE REVIEW

File: .travis/environment.yaml [hunk 1]
```diff
@@ -13,18 +13,17 @@ dependencies:
...
 - jsonschema
 - matplotlib # pixel drill app
 - mock # testing
 - netcdf4
-- numexpr # For AE/EE
 - numpy
 - paramiko # for simple-replicas
 - pathlib
 - pathos
+- pep8 # testing
```
Comment (jeremyh) [inline .travis/environment.yaml]: I think we replaced `pep8` everywhere with pycodestyle?

File: datacube/scripts/ingest.py [hunk 2]
```diff
@@ -53,6 +53,9 @@ def morph_dataset_type(source_type, config, index, storage_format):
 
     output_type.metadata_doc['format'] = {'name': storage_format}
 
+    if 'metadata_type' in config:
+        output_type.definition['metadata_type'] = config['metadata_type']
+
```
Comment (jeremyh) [inline datacube/scripts/ingest.py]: Merge bug? Duplicates the block below it

**Review by jeremyh — DISMISSED** [review]
:+1: Thanks for the doing the merging

**Review by andrewdhicks — DISMISSED** [review]

**Review by andrewdhicks — APPROVED** [review]
+1

**Pull request merged**